rfd/0057-automatic-aws-server-discovery.md

```diff
@@ -235,6 +235,153 @@ discovery_service:
     teleport.dev/origin: "cloud"
 ```
 
+
+### Agentless installation
+
+In addition to supporting automatic Teleport agent installation, an agentless option
+will also be supported. This mode will update the OpenSSH CA to use the Teleport CA
+without installing the full Teleport Agent.
+
+The teleport ca and host keys will be generated and then uploaded to
+the the AWS Secret store where, for each node discovered a new secret
+will be created, each secret will have the following contents
+
+```json
+{
+	"ca_key": "...",
+	"host_key": "...",
+	"host_cert": "...",
+}
+```
+
+This will require changes to Teleport in order to support RBAC of OpenSSH based
+nodes and to support showing the nodes in tsh/web ui.
+
+This mode can be enabled by setting `agentless: true` in the matcher. When the
+matcher includes this, a predefined script for agentless installation will be used for
+the endpoint.
+
+Example agentless config:
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  aws:
+  - types: ["ec2"]
+    regions: ["us-west-1"]
+    tags:
+      "teleport": "yes" # aws tags to match
+    install:
+      agentless: true
```

Comment:
@klizhentas @xinding33 What do you think about this way to toggle the auto-discovery mode where we don't install agents on the nodes? The "agentless" toggle feels a little awkward to me and I wonder if it may be confusing for users.

If we could go back in time, I would probably prefer that we started with "agentless" mode by default and had a setting like `install_teleport: true` (i.e. inverted). We can probably still do it and just have `install_teleport` default to "true" for now (to keep backwards compatibility) and if we wanted to make agentless default, bump our config version again.

What do you think? About agentless/non-agentless mode as default and the way to pick one vs another.

Reply:
why don't we add `install_teleport: false` and make `true` the default in the current version of the config, and bump the version to v++